src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -66,6 +66,16 @@ CascadeOperationForConnectedRelations(Oid relationId, LOCKMODE lockMode,
 	InvalidateForeignKeyGraph();
 
 	List *fKeyConnectedRelationIdList = GetForeignKeyConnectedRelationIdList(relationId);
+
+	/*
+	 * If relation is not involved in any foreign key relationships, foreign
```

Comment:
why don't we follow the same approach with CitusLocalTable such that we only get into cascade logic if `TableHasExternalForeignKeys()` ?
```C
bool tableHasExternalForeignKeys = TableHasExternalForeignKeys(relationId);
	if (tableHasExternalForeignKeys && cascadeViaForeignKeys)
```


That feels more natural/expected to me